utils/utils.hpp

```diff
@@ -329,5 +329,28 @@ class Utils
       mlpack::Log::Warn << "The " << path << " doesn't exist." << std::endl;
     }
   }
+
+  /**
+   * Loads a vocabulary file and stores the content into a vector.
+   *
+   * @param vocabPath The path to the vocabulary file.
+   * @param vocabulary Stores the vocabulary content along with indices.
+   */
+  void LoadVocabulary(const std::string vocabPath,
+                      std::map<std::string, size_t>& vocabulary)
+  {
+    std::string token;
+    std::ifstream vocabFile(vocabPath);
+    if (vocabFile.is_open())
+    {
+      for (size_t i = 0; std::getline(vocabFile, token); ++i)
+        vocabulary[token] = i;
+
+      vocabFile.close();
```

Comment:
You don't need to close the file explicitly since it'll be closed in the destructor.